From 6103c916f18959121cc0a0add4df5b05161f418d Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Mon, 17 Aug 2020 13:44:26 +0530
Subject: [PATCH 38/38] Add-G133HAN01-Dual-Channel-LVDS-Support

This patch will add support for G133HAN01 Dual channel
LVDS support and uses "jeida" as data-mapping format.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 49 +++++++++++++++++++++++++--------
 arch/arm/boot/dts/lec-imx6q.dts |  4 +++
 arch/arm/boot/dts/lec-imx6s.dts |  4 +++
 3 files changed, 45 insertions(+), 12 deletions(-)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index 2aee16d91048..2188564a8318 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -257,7 +257,7 @@
 		pwms = <&pwm1 0 5000000>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
 		default-brightness-level = <6>;
-		enable-gpios = <&gpio6 10 GPIO_ACTIVE_HIGH>; /* SMARC LCD_BKLT_EN */
+		enable-gpios = <&gpio3 31 GPIO_ACTIVE_HIGH>; /* SMARC LCD_BKLT_EN */
 		status = "okay";
 	};
 
@@ -1010,6 +1010,7 @@
 			fsl,pins = <
 				MX6QDL_PAD_NANDF_RB0__GPIO6_IO10	0x80000000 /* SMARC LCD_BKLT_EN */
 				MX6QDL_PAD_GPIO_9__PWM1_OUT		0x1b0b1 /* SMARC LCD_BKLT_PWM */
+				MX6QDL_PAD_EIM_D31__GPIO3_IO31		0x80000000 /* SMARC LCD_BKLT_EN (iMX6R2) */
 			>;
 		};
 
@@ -1118,29 +1119,53 @@
 };
 
 &ldb {
+	disp-power-on-supply = <&reg_smarc_lcd_vdd>; /* unsupported by driver */
+	split-mode;
 	status = "okay";
 
 	lvds-channel@0 {
-		fsl,data-mapping = "spwg";	/* spwg or jeida (see chapter "Bit Mapping" in i.MX6 Reference Manual) */
+		fsl,data-mapping = "jeida";	/* spwg or jeida (see chapter "Bit Mapping" in i.MX6 Reference Manual) */
 		fsl,data-width = <24>;
 		primary;
 		status = "okay";
 
 		display-timings {
 			native-mode = <&timing0>;
-			timing0: hsd100pxn1 {
-				clock-frequency = <65000000>;
-				hactive = <1024>;
-				vactive = <600>;
-				hback-porch = <220>;
-				hfront-porch = <40>;
-				vback-porch = <21>;
-				vfront-porch = <7>;
-				hsync-len = <60>;
-				vsync-len = <10>;
+			timing0: g133han01 {
+				clock-frequency = <141200000>;
+				hactive = <1920>;
+				vactive = <1080>;
+				hback-porch = <88>;
+				hfront-porch = <58>;
+				vback-porch = <14>;
+				vfront-porch = <8>;
+				hsync-len = <42>;
+				vsync-len = <14>;
 			};
 		};
 	};
+
+	lvds-channel@1 {
+		fsl,data-mapping = "jeida";     /* spwg or jeida (see chapter "Bit Mapping" in i.MX6 Reference Manual) */
+		fsl,data-width = <24>;
+		primary;
+		status = "okay";
+
+		display-timings {
+			native-mode = <&timing1>;
+			timing1: g133han01 {
+				clock-frequency = <141200000>;
+				hactive = <1920>;
+				vactive = <1080>;
+				hback-porch = <88>;
+				hfront-porch = <58>;
+				vback-porch = <14>;
+				vfront-porch = <8>;
+				hsync-len = <42>;
+				vsync-len = <14>;
+                        };
+                };
+        };
 };
 
 &pcie {
diff --git a/arch/arm/boot/dts/lec-imx6q.dts b/arch/arm/boot/dts/lec-imx6q.dts
index 1c2e43aaf255..b1066329e996 100644
--- a/arch/arm/boot/dts/lec-imx6q.dts
+++ b/arch/arm/boot/dts/lec-imx6q.dts
@@ -18,6 +18,10 @@
 	lvds-channel@0 {
 		crtc = "ipu2-di0";
 	};
+
+	lvds-channel@1 {
+		crtc = "ipu2-di1";
+	};
 };
 
 &sata {
diff --git a/arch/arm/boot/dts/lec-imx6s.dts b/arch/arm/boot/dts/lec-imx6s.dts
index a91c2c57a07a..30d0ec3f772b 100644
--- a/arch/arm/boot/dts/lec-imx6s.dts
+++ b/arch/arm/boot/dts/lec-imx6s.dts
@@ -24,6 +24,10 @@
 	lvds-channel@0 {
 		crtc = "ipu1-di0";
 	};
+
+	lvds-channel@1 {
+		crtc = "ipu2-di1";
+	};
 };
 
 #if CONFIG_SOLO_LVDS
-- 
2.17.1

